src/makeInfinityAdder.js

```diff
@@ -1,12 +1,26 @@
 'use strict';
 
 /***
- *
  * @return {function}
  */
-
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(number) {
+    if (number === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    } else {
+      sum += number;
+
+      return adder;
```

Comment:
Ensure that the input `number` is always a number or undefined. Currently, the function will incorrectly add other types like strings or objects if they are provided. You might want to explicitly check the type of `number` and throw an error if it is not a number or undefined.